src/runtime/pkg/containerd-shim-v2/device_cold_plug.go

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2025 NVIDIA CORPORATION.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package containerdshim
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"runtime"
+
+	"github.com/kata-containers/kata-containers/src/runtime/pkg/device/config"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	podresourcesv1 "k8s.io/kubelet/pkg/apis/podresources/v1"
+)
+
+const (
+	nameAnnotation      = "io.kubernetes.cri.sandbox-name"
+	namespaceAnnotation = "io.kubernetes.cri.sandbox-namespace"
+)
+
+// coldPlugDevices handles cold plug of CDI devices into the sandbox
+// kubelet's PodResources API is used for determining the devices to be
+// cold plugged, if so configured. Otherwise, cdi annotations can be used for
+// covering stand alone scenarios.
+func coldPlugDevices(ctx context.Context, s *service, ociSpec *specs.Spec) error {
+	if !s.config.ColdPlugDevices {
+		// device cold plug is not enabled
+		return nil
+	}
+
+	kubeletSock := s.config.PodResourceAPISock
+	if kubeletSock != "" {
+		return coldPlugWithAPI(ctx, s, ociSpec)
+	}
```

Comment:
Shall we add a clear debug message here that we're not attemping the k8s based coldplug in case kubeletSock is not set? (Yep, it's mentioned in the configuration file that this would be case, but this can help us when debugging it in the future)